packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/IrUtils.kt

```diff
@@ -125,7 +125,24 @@ inline fun ClassDescriptor.hasInterfacePsi(interfaces: Set<String>): Boolean {
     this.findPsi()?.acceptChildren(object : PsiElementVisitor() {
         override fun visitElement(element: PsiElement) {
             if (element.node.elementType == SUPER_TYPE_LIST) {
-                hasRealmObjectAsSuperType = element.node.text.findAnyOf(interfaces) != null
+                // Check supertypes for classes with Embbeded/RealmObject as generics and remove
+                // them from the string so as to avoid erroneously processing said classes
+                // implementing these types as implementing Embedded/RealmObject. Doing so would
+                // add our companion interface causing compilation errors.
+                val elementNodeText = element.node.text
+                    .replace(" ", "") // Sanitize removing spaces
+                    .split(",") // Split by commas
+                    .mapNotNull {
```

Comment:
Seems like a job for `filter` 🤪

Reply:
Yeah, it was my first attempt, I'll try to improve later today